app/scripts/modules/core/src/application/nav/ApplicationNavigation.tsx

```diff
@@ -51,11 +51,8 @@ export const ApplicationNavigation = ({ app }: IApplicationNavigationProps) => {
   });
 
   React.useEffect(() => {
-    appRefreshSubscription();
-
-    return () => {
-      appRefreshSubscription();
-    };
+    const unsubscribe = appRefreshSubscription();
+    return unsubscribe;
```

Comment:
I assume you separated these into two statements so it's more explicit what happens here?

It might be even more better to show an arrow doing an action:

```suggestion
    const unsubscribe = appRefreshSubscription();
    return () =>  unsubscribe();
```

no big deal if you don't like 

FWIW I am pretty sure we have code elsewhere in deck written like this: 
```
React.useEffect(() => appRefreshSubscription(), []);
```
but that is definitely less apparent